src/server/admin/cmds/cmds.go

```diff
@@ -63,6 +69,33 @@ $ {{alias}} -u s3://bucket/backup`,
 			if err != nil {
 				return err
 			}
+
+			// Generate a root auth token and cache it in the current context. If the restore operation
+			// activates auth this will become the new root token.
+			if !noToken {
+				authToken := uuid.NewWithoutDashes()
```

Comment:
One arguable benefit of this approach that occurred to me is that if you restore the same dump twice into two different clusters, they'll have different root tokens (which will strengthen the invariant that all clusters have a unique root token)

Reply:
Yeah I had this thought as well. Sharing tokens between clusters and the risk of losing a token between extracting and restoring seem to outweigh the danger of a partial restore.